### DISCUSSION TITLE
Bulk-Delete short links based on tags

### DISCUSSION THREAD

**Discussion by SoCuul** [shlinkio/discussions#113] [Help wanted] [answered]
#### How Shlink is set up

* Shlink Version: 3.5.4
* PHP Version: 8.2.4
* How do you serve Shlink: RoadRunner Docker image
* Database engine used: PostgreSQL 12.2.1

#### Summary

I'm wondering if it's possible to bulk-delete all short links based on the included tags.
For example, I would want to delete all links that have the tag "summer-lto".
Would I have to do this manually through SQL, or is there a better way?

Thanks

**Comment by acelaya** [comment, marked as answer]
There's not one endpoint to do this in one go, if that's what you mean, but you could combine a couple of them to get it scripted without having to deal with SQL, which would risk ending up with data inconsistency,

* First list all short URLs having that specific tag: https://api-spec.shlink.io/?version=v3.5.4#/Short%20URLs/listShortUrls
* Then iterate over them, and call the delete endpoint for each one: https://api-spec.shlink.io/?version=v3.5.4#/Short%20URLs/deleteShortUrl
